### PR TITLE
Grpc.Tools: Use x64 protoc on macOS arm64

### PR DESCRIPTION
**Pull request by BradBarnich**
As a workaround until #25755 is resolved.

@markdroth

### CODE REVIEW

File: src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets [hunk 1]
```diff
@@ -75,6 +75,10 @@
       <!-- Next try OS and CPU resolved by ProtoToolsPlatform. -->
       <Protobuf_ToolsOs Condition=" '$(Protobuf_ToolsOs)' == '' ">$(_Protobuf_ToolsOs)</Protobuf_ToolsOs>
       <Protobuf_ToolsCpu Condition=" '$(Protobuf_ToolsCpu)' == '' ">$(_Protobuf_ToolsCpu)</Protobuf_ToolsCpu>
+ 
+      <!-- Use x64 on macosx arm64 until a native protoc is shipped -->
+      <Protobuf_ToolsCpu Condition=" '$(Protobuf_ToolsCpu)' == 'arm64' and '$(Protobuf_ToolsOs)' == 'macosx' ">x64</Protobuf_ToolsCpu>
```
Comment (jtattermusch) [inline src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets:80]: If you want to override the cpu architecture, you should do it in the C# code
(e.g. in the https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Tools/ProtoToolsPlatform.cs), not here.

The msbuild files already way too complex, they weren't designed for hacks like this and they are also much harder to test than the C# code.

**Review by jtattermusch — CHANGES_REQUESTED** [review]
See the comments in the code.

**Comment by BradBarnich** [comment]
@jtattermusch Thank you, can you give this new version a look?

**Comment by jtattermusch** [comment]
I started an adhoc run of  package build and distribtest: https://fusion2.corp.google.com/invocations/6aedd5f1-57dc-443f-a335-f2320090f7e2/targets

I will post results later.

**Review by jtattermusch — APPROVED** [review]
LGTM once the tests are green and an adhoc build of Grpc.Tools is tested.

**Comment by jtattermusch** [comment]
@apolcyn  please add another Gooogler's review.

**Comment by jtattermusch** [comment]
One more run of distribtests:
https://fusion2.corp.google.com/invocations/d7c3d66c-f92c-4a78-9bba-75f24fcde425/targets

**Review by apolcyn — APPROVED** [review]

**Comment by jtattermusch** [comment]
Distribtests are green: https://fusion2.corp.google.com/invocations/7a23e587-b8cc-4512-a8dd-8ca0bcaecef3 (except of distribtest.python_linux_x64_ubuntu1604, which is unrelated and is being fixed).

**Pull request merged**